csrc/executor_params.cpp

```diff
@@ -42,18 +82,32 @@ void LaunchParams::bind(int64_t val, ParallelType p_type) {
     case ParallelType::BIDx:
       checkAndSet(val, gdimx_, "gridDim.x");
       break;
+    // current implementation require: gridDim.{xyz} = clusterDim.{xyz} if
+    // clusterDim.{xyz} > 1
```

Comment:
Am I understanding correct that `clusterDim.x == gridDim.x` means the whole grid is in one cluster, and `clusterDim.x == 1` means that each cluster has 1 CTA on the x direction? Could you add assert for your assumption in `assertValid`?

Reply:
(1) It means all the blocks in the x-dim of the grid is one cluster. See `ThreadBlockClusterTest.OneClusterEachRow`.
(2) `clusterDim.x` means number of CTAs in this cluster in x direction. So `clusterDim.x == 1` means that each cluster has 1 CTA on the x direction.